blog/_posts/2017-11-30-bloop-release.md

```diff
@@ -0,0 +1,153 @@
+faster than a hot
+compiler](https://scala-ci.typesafe.com/grafana/dashboard/db/scala-benchmark?var-branch=2.12.x&var-source=vector&var-bench=HotScalacBenchmark.compile&var-host=scalabench@scalabench@).
+
+Keeping a hot compiler in the background is crucial if one takes compilation speed and productivity
+seriously. This is often difficult and takes space of mind. For example, `sbt` throws away a hot
+compiler every time users quit (`Ctrl-C`) and restart, but also when they do `reload` after several
+executions of `compile`. By this rule, the more users modify their sbt build, the less productive
+they are.
+
+Current sbt users can indeed avoid quitting and prefer suspending (`Ctrl-Z`), but they will always
+be punished for modifying their builds. In our experience, changing habits is hard, so we have
+preferred to add a built-in nailgul integration in `bloop` that addresses this problem at its root
+rather than changing people's habits, which is hard and time consuming.
+
+### How does it work?
+
+When `bloop` starts up, it loads a build from a configuration directory passed in by the user, which
+defaults to `.bloop-config`. This directory contains a configuration file per every project, and
+every configuration file contains basic information about the project: name, base directory,
+classpath, source directories, classes directory, Scala and Java options and a few more, which are
+hard to create and maintain.
+
+`bloop` comes with an sbt plugin that allows you to generate these configuration files with the
+execution of a command. When the command is done, you can run your build with `bloop`.
+
+This configuration file is a properties file that we plan on migrating from. In the next weeks, we'd
+like to come up with the design of a configuration file that we can reuse across different tools in
+the Scala ecosystem. Given how common this kind of information is required, we believe that settling
+on an agreed upon configuration format is the way to go.
+
+In the future, we want to explore making `bloop` a server you can ask compilation and test-related
+information from different language servers, presentation compilers and linters.
+
+## Numbers on compilation performance
+
+`bloop` has been created out of the observation that Scala compile times slow down within `sbt` in
+comparison with isolated benchmarks. This observation was at the beginning just an intuition, but
+when we sat down to measure a prototype we did see a significant difference.
+
+<blockquote class="twitter-tweet" data-lang="en-gb"><p lang="en" dir="ltr"> Scala Tooling has got to
+a point where people say "compiler performance" and what they really mean is "sbt performance".
+</p>&mdash; Jorge (@jvican) <a href="https://twitter.com/jvican/status/928601470232129536">9th
+November 2017</a></blockquote><script async src="//platform.twitter.com/widgets.js"
+charset="utf-8"></script>
+
+We'd like to share some of our numbers with you.
```

Comment:
What are these numbers describing? You should add a paragraph explaining what you benchmarked!